app/common/lib/bookmarkUtil.js

```diff
@@ -65,8 +65,18 @@ const getDNDBookmarkData = (state, bookmarkKey) => {
   return data.get('draggingOverKey') === bookmarkKey ? data : Immutable.Map()
 }
 
+let oldSites
+let lastValue
+let lastWidth
 const getToolbarBookmarks = (state) => {
   const sites = state.get('sites', Immutable.List())
+  if (sites === oldSites &&
+      lastWidth === window.innerWidth &&
```

Comment:
window width is available in appstate.windows, but at the very least we should call this once and save the value because each call forces a layout recalc in webkit